src/io/hdf/hdf_path.cpp

```diff
@@ -23,8 +23,18 @@ hdf_path& hdf_path::append(const hdf_path& p)
 {
   if (p.has_root_directory())
     path_.clear();
-  else if (path_.back() != '/')
-    path_.push_back('/');
+  else
+  {
+    if (path_.size() > 0)
+    {
+      if (path_.back() != '/')
+        path_.push_back('/');
+    }
+    else
+    {
+      path_.push_back('/');
```

Comment:
```suggestion
    if (path_.empty())
      path_.push_back('/');
    else if (path_.back() != '/')
      path_.push_back('/');
```
Is this code more clear?

Reply:
If `p` is an absolute path, it overwrites the string.

if `path_` is empty, we add `/` to create an absolute path and call concat.

otherwise we check to avoid a `//` and call concat.

Reply:
I tried to follow the same rules as `std::filesystem::path::append`
https://en.cppreference.com/w/cpp/filesystem/path/append

Reply:
Both clearer and more concise, at least to me. I have pushed the change & verified it is OK locally, as expect.

Reply:
> if path_ is empty, we add / to create an absolute path and call concat.

I'm not following this rule. If `path_` is empty and a user appends `p` which is a relative path, I don't see why adding a root `/` is valid.

Reply:
```
If p.is_absolute() || (p.has_root_name() && p.root_name() != root_name()), then replaces the current path with p as if by operator=(p) and finishes.
```
https://en.cppreference.com/w/cpp/filesystem/path/append

Reply:
> ```
> If p.is_absolute() || (p.has_root_name() && p.root_name() != root_name()), then replaces the current path with p as if by operator=(p) and finishes.
> ```
> 
> https://en.cppreference.com/w/cpp/filesystem/path/append

not following. in my question, `p` is not absolute and has no root_name.

Reply:
I think you're right that `hdf_path` isn't following `std::filesystem::path` in that case. Hopefully nothing depends on this behavior and it's an easy change :crossed_fingers: 

std::filesystem::path::append doesn't add a `/` at the beginning.  
https://godbolt.org/z/981zhK4sv